core/coins/format_test.go

```diff
@@ -0,0 +1,81 @@
+package coins_test
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+
+	bankv1beta1 "cosmossdk.io/api/cosmos/bank/v1beta1"
+	basev1beta1 "cosmossdk.io/api/cosmos/base/v1beta1"
+	"cosmossdk.io/core/coins"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFormatCoin(t *testing.T) {
+	var testcases []coinJsonTest
+	raw, err := os.ReadFile("../../tx/textual/internal/testdata/coin.json")
+	require.NoError(t, err)
+	err = json.Unmarshal(raw, &testcases)
+	require.NoError(t, err)
+
+	for _, tc := range testcases {
+		t.Run(tc.Text, func(t *testing.T) {
+			if tc.Proto != nil {
+				out, err := coins.FormatCoins([]*basev1beta1.Coin{tc.Proto}, []*bankv1beta1.Metadata{tc.Metadata})
+
+				if tc.Error {
+					require.Error(t, err)
+					return
+				}
+
+				require.NoError(t, err)
+				require.Equal(t, tc.Text, out)
+			}
+		})
+	}
+}
+
+func TestFormatCoins(t *testing.T) {
+	var testcases []coinsJsonTest
+	raw, err := os.ReadFile("../../tx/textual/internal/testdata/coins.json")
```

Comment:
This `core` test imports a `tx` module's json file, which is not ideal.

We could alternatively:
- test FormatCoins in the `tx` module. Con: unit test not next to its function
- move coins.json into core. Con: we'll have some ValueRenderer json files in `tx`, others in `core`

For now the PR's version is my favorite. Any other opinions?

Reply:
I would say use `go/embed` + https://github.com/golang/go/wiki/Modules#can-a-module-depend-on-an-internal-in-another

Reply:
What you're doing is also maybe not terrible 🤷

Reply:
I am going to keep it like this for now